integration/emr/alluxio-emr.sh

```diff
@@ -10,99 +10,187 @@
-fi
-
-WORKERS=`printf '%s\n' "${HOSTLIST//$MASTER/}"`
-
-sudo runuser -l alluxio -c "echo '$MASTER' > /opt/alluxio/conf/masters"
-sudo runuser -l alluxio -c "echo '$WORKERS' > /opt/alluxio/conf/workers"
-
-#Identify master
-IS_MASTER=`jq '.isMaster' /mnt/var/lib/info/instance.json`
-
-#Set up alluxio-site.properties
-sudo runuser -l alluxio -c "echo 'alluxio.master.hostname=$MASTER' > /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.master.journal.type=UFS' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.master.mount.table.root.ufs=$2' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.master.security.impersonation.hive.users=*' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.master.security.impersonation.presto.users=*' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.master.security.impersonation.yarn.users=*' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.worker.memory.size=20GB' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.worker.tieredstore.level0.alias=MEM' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.worker.tieredstore.level0.dirs.path=/mnt/ramdisk' >> /opt/alluxio/conf/alluxio-site.properties"
-sudo runuser -l alluxio -c "echo 'alluxio.worker.tieredstore.levels=1' >> /opt/alluxio/conf/alluxio-site.properties"
-
-#Inject user defined properties (semicolon separated)
-IFS=';'
-conf=($3)
-printf "%s\n" "${conf[@]}" | sudo tee -a /opt/alluxio/conf/alluxio-site.properties
-
-#No ssh
-if [[ ${IS_MASTER} = "true" ]]
-then
-  sudo runuser -l alluxio -c "/opt/alluxio/bin/alluxio-start.sh -a master"
-  sudo runuser -l alluxio -c "/opt/alluxio/bin/alluxio-start.sh -a job_master"
-  sudo runuser -l alluxio -c "/opt/alluxio/bin/alluxio-start.sh -a proxy"
-else
-  /opt/alluxio/bin/alluxio-mount.sh SudoMount local
-  until /opt/alluxio/bin/alluxio fsadmin report
-  do
-    sleep 5
+
+  local release=`basename ${alluxio_tarball}`
+  local release_unzip=${release%"-bin.tar.gz"}
+  release_unzip=${release_unzip%".tar.gz"}
+
+  # Unpack and inflate the release tar
+  # TODO logic for different compression formats, s3 URIs, git URIs, etc.
+  sudo cp ${release} /opt/
+  sudo tar -xvf /opt/${release} -C /opt/
+  sudo rm -R /opt/${release}
+  sudo mv /opt/${release_unzip} ${ALLUXIO_HOME}
+  sudo chown -R alluxio:alluxio ${ALLUXIO_HOME}
+  rm ${release}
+  doas alluxio "cp ${ALLUXIO_SITE_PROPERTIES}.template ${ALLUXIO_SITE_PROPERTIES}"
+
+  #Get hostnames and load into masters/workers file
+  local emr_cluster=`jq '.jobFlowId' /mnt/var/lib/info/job-flow.json | sed -e 's/^"//' -e 's/"$//'`
+  local hostlist=`aws emr list-instances --cluster-id ${emr_cluster} --region us-east-1 | jq '.Instances[].PrivateDnsName' | sed -e 's/^"//' -e 's/"$//'`
```

Comment:
Will eventually need to figure out a way to programmatically determine region. Current it is hardcoded to `us-east-1`

Reply:
Fixed in later commit